projects/kit/components/select/select-option/select-option.component.ts

```diff
@@ -0,0 +1,63 @@
+import {ChangeDetectionStrategy, Component, computed, inject} from '@angular/core';
+import {TuiControl} from '@taiga-ui/cdk/classes';
+import {tuiDirectiveBinding, tuiIsPresent} from '@taiga-ui/cdk/utils/miscellaneous';
+import type {TuiDataListHost} from '@taiga-ui/core/components/data-list';
+import {
+    TUI_DATA_LIST_HOST,
+    TuiOptionWithValue,
+} from '@taiga-ui/core/components/data-list';
+import {TuiScrollIntoView} from '@taiga-ui/core/components/scrollbar';
+import {tuiInjectAuxiliary} from '@taiga-ui/core/components/textfield';
+import {TuiIcons} from '@taiga-ui/core/directives/icons';
+import {TuiItemsHandlersDirective} from '@taiga-ui/core/directives/items-handlers';
+import {TUI_COMMON_ICONS} from '@taiga-ui/core/tokens';
+
+@Component({
+    standalone: true,
+    selector: 'tui-select-option',
+    template: '',
+    styleUrls: ['./select-option.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    hostDirectives: [TuiScrollIntoView],
+    host: {
+        '[class._value]': 'option',
+        '[class._selected]': 'selected()',
+        '(click)': 'onClick()',
+    },
+})
+export class TuiSelectOption<T> {
+    private readonly host = inject<TuiDataListHost<T>>(TUI_DATA_LIST_HOST);
+    private readonly itemsHandlers = inject(TuiItemsHandlersDirective<T>);
+    private readonly control = tuiInjectAuxiliary<TuiControl<T>>(
+        (x) => x instanceof TuiControl,
+    );
+
+    protected readonly option = inject<TuiOptionWithValue<T>>(TuiOptionWithValue, {
+        optional: true,
+    });
+
+    protected readonly icon =
+        this.option &&
+        tuiDirectiveBinding(TuiIcons, 'iconEnd', inject(TUI_COMMON_ICONS).check, {});
```

Comment:
Do we need dependence on the size here? I believe in proprietary we have different checkmark icon for 16 and 24 sizes.

Reply:
You propose to stop using `TUI_COMMON_ICONS`-token for new selects and implement the following logic, don't you?
```ts
// select.options.ts
export const TUI_SELECT_DEFAULT_OPTIONS: TuiSelectOptions = {
    icon: () => '@tui.check',
};
```

```ts
// select-option.component.ts
protected readonly icon = this.option && tuiTextfieldIconBinding(TUI_SELECT_OPTIONS);
```

Reply:
Discuss personally it with @waterplea – we decide to leave it as it is. Maybe we will change it later.